tools/indexer/wallet/src/main.rs

```diff
@@ -0,0 +1,221 @@
+            account_id: account_id.to_string(),
+            action: ADD,
+            status: SUCCESS,
+            receipt_hash: "genesis".to_string(),
+            block_height: BigDecimal::from_u64(genesis_height).expect("genesis_height must be provided"),
+            permission: match access_key.permission {
+                near_indexer::near_primitives::account::AccessKeyPermission::FullAccess => FULL_ACCESS,
+                near_indexer::near_primitives::account::AccessKeyPermission::FunctionCall(_) => FUNCTION_CALL,
+            }
+        })
+    } else { None }).collect::<Vec<AccessKey>>();
+
+    diesel::insert_into(schema::access_keys::table)
+        .values(access_keys)
+        .on_conflict_do_nothing()
+        .execute_async(&pool)
+        .await
+        .unwrap();
+}
+
+async fn listen_blocks(mut stream: mpsc::Receiver<near_indexer::BlockResponse>) {
+    let pool = establish_connection();
+
+    while let Some(block) = stream.recv().await {
+        eprintln!("Block height {:?}", block.block.header.height);
+
+        // Handle outcomes
+        let receipt_outcomes = &block
+            .outcomes
+            .iter()
+            .filter_map(|outcome| match outcome {
+                near_indexer::Outcome::Receipt(execution_outcome) => Some(execution_outcome),
+                _ => None,
+            })
+            .collect::<Vec<&near_indexer::near_primitives::views::ExecutionOutcomeWithIdView>>();
+
+        diesel::update(
+            schema::access_keys::table.filter(
+                schema::access_keys::dsl::receipt_hash.eq_any(
+                    receipt_outcomes
+                        .iter()
+                        .filter(|outcome| match &outcome.outcome.status {
+                            near_indexer::near_primitives::views::ExecutionStatusView::Failure(
+                                _,
+                            ) => true,
+                            _ => false,
+                        })
+                        .map(|outcome| outcome.id.clone().to_string())
+                        .collect::<Vec<String>>(),
+                ),
+            ),
+        )
+        .set(schema::access_keys::dsl::status.eq(FAILED))
+        .execute_async(&pool)
+        .await
+        .unwrap();
+
+        diesel::update(
+            schema::access_keys::table.filter(
+                schema::access_keys::dsl::receipt_hash.eq_any(
+                    receipt_outcomes
+                        .iter()
+                        .filter(|outcome| match &outcome.outcome.status {
+                            near_indexer::near_primitives::views::ExecutionStatusView::SuccessReceiptId(
+                                _,
+                            ) | near_indexer::near_primitives::views::ExecutionStatusView::SuccessValue(_) => true,
+                            _ => false,
+                        })
+                        .map(|outcome| outcome.id.clone().to_string())
```

Comment:
why not include failed ones here?

Reply:
Failed ones updating a bit above in the code. Separately updating all the failing with status and then all the succeeded.